pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -200,3 +297,143 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 	)
 	return nil
 }
+
+func buildKubeClientWithSecret(secret *corev1.Secret) (kubernetes.Interface, error) {
+	defer func() {
+		for key := range secret.Data {
+			os.Remove(path.Join("/tmp", key))
+		}
+	}()
+
+	for key, data := range secret.Data {
+		if err := ioutil.WriteFile(path.Join("/tmp", key), data, 0644); err != nil {
+			return nil, err
+		}
+	}
+	restConfig, err := clientcmd.BuildConfigFromFlags("", path.Join("/tmp", "kubeconfig"))
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(restConfig)
+}
+
+func createSelfSubjectAccessReviews(
+	ctx context.Context,
+	kubeClient kubernetes.Interface,
+	selfSubjectAccessReviews []authorizationv1.SelfSubjectAccessReview) (bool, *authorizationv1.SelfSubjectAccessReview, error) {
+	for _, subjectAccessReview := range selfSubjectAccessReviews {
+		ssar, err := kubeClient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, &subjectAccessReview, metav1.CreateOptions{})
```

Comment:
The go `range` statement strikes again! It's not safe to use pointers to variables like `subjectAccessReview` that are created by the range statement. When you take a pointer to this, you are taking a pointer to _the variable you're using in the loop_, not the object in the collection. So, this will likely not work as you might expect.

Instead, you need to use this construction:

```go
for i := range selfSubjectAccessReviews {
  sar := selfSubjectAccessReviews[i]

  // use sar
  thing(&sar)
}
```

[more info](https://tam7t.com/golang-range-and-pointers/) here

Reply:
fixed